src/data/polynomial/degree/definitions.lean

```diff
@@ -640,6 +648,14 @@ lemma monic.ne_zero_of_ne (h : (0:R) ≠ 1) {p : R[X]} (hp : p.monic) :
   p ≠ 0 :=
 by { nontriviality R, exact hp.ne_zero }
 
+lemma monic_of_nat_degree_le_of_coeff_eq_one (pn : p.nat_degree ≤ n) (p1 : p.coeff n = 1) :
+  monic p :=
+begin
+  nontriviality,
+  refine (congr_arg _ $ pn.antisymm $ le_nat_degree_of_ne_zero _).trans p1,
```

Comment:
You can now use `nat_degree_eq_of_le_of_coeff_ne_zero pn _` here, right?

Reply:
Yes, I did in the new version.  It is a little disappointing that the lemma names are so long that, even though they short-circuit a couple of steps, the actual number of characters does not decrease.